crates/router/src/utils/user_role.rs

```diff
@@ -255,12 +255,54 @@ pub async fn get_lineage_for_user_id_and_entity_for_accepting_invite(
 ) -> UserResult<
     Option<(
         id_type::OrganizationId,
-        id_type::MerchantId,
+        Option<id_type::MerchantId>,
         Option<id_type::ProfileId>,
     )>,
 > {
     match entity_type {
-        EntityType::Organization => Err(UserErrors::InvalidRoleOperation.into()),
+        EntityType::Organization => {
+            let org_id = id_type::OrganizationId::try_from(std::borrow::Cow::from(entity_id))
+                .change_context(UserErrors::InternalServerError)?;
```

Comment:
Can we have a context message here (with `attach_printable` / `attach_printable_lazy`) and other places where we're raising internal server errors?

Reply:
- I've gone through the code and checked wherever we are throwing `InternalServerError`. All of the places are mostly
  1. Data problems, which we can find using DB efficiently as we don't have the whole data of the object.
  2. Logical errors or Unreachable errors, which can be found out if we see the code at the line number at which the error is logged.
- In my opinion, adding `attach_printable`s without any important debug information will pollute the code. So, that's why I don't prefer adding them unless they are adding any value to the log.
- In the place where you commented, there was no need to throw 500 and I changed that code.